map_test.go

```diff
@@ -433,6 +433,32 @@ func TestMapPinWithEmptyPath(t *testing.T) {
 	c.Assert(err, qt.Not(qt.IsNil))
 }
 
+func TestMapPinReplace(t *testing.T) {
```

Comment:
Could you add a quick comment here describing which error case is being tested? It's not immediately clear what this code is supposed to do, and the name of the test isn't descriptive. (perhaps adding a `qt.Commentf()` to the final assertion might help)